src/sagas.js

```diff
@@ -809,7 +813,20 @@ export function* fetchVmSnapshots ({ vmId }) {
   let snapshotsInternal = []
 
   if (snapshots && snapshots['snapshot']) {
-    snapshotsInternal = snapshots.snapshot.map(snapshot => Api.snapshotToInternal({ snapshot }))
+    snapshotsInternal = yield all(snapshots.snapshot.map(snapshot =>
+      call(function* () {
```

Comment:
I suggest putting a `NOTE` comment here saying something like "Snapshot Disks and Nics are not currently available via additional/follow param on the snapshot fetch so we need to fetch them directly".

I also recommend filing a BZ RFE to add that capability, and an issue to track it.  There are a bunch of places I've seen like that where it would be great to not have to make additional calls.  Might as well start logging RFE gaps now right?  Or are these the types of things Mike is going to fix?  Either way, a BZ would help track the change in engine project...

cc: @gregsheremeta

Reply:
I teased it apart a bit to a few different functions so it is easier to read.  Applied the same work to `fetchSingleVm`.